Makefile.am

```diff
@@ -1,6 +1,2 @@
-if SET_DLSCHED
```

Comment:
This can't be removed unconditionally.
We build libdl only if libc doesn't support sched_setattr.

Reply:
I remember in our call when Patrick was here we agreed to always compile deadline in as it's been around for long enough now. Am I mis-remembering?

Reply:
Yes it's right, you don't mis-remember but SET_DLSCHED is not about compiling the support of deadline scheduler but about compiling a local support of sched_setattr if the lib doesn't support it.
So DLSCHED must be removed but not SET_DLSCHED

Reply:
Ah I see. Okay let me see if I can hack something and updated the PR

Reply:
It should be just keeping this : if SET_DLSCHED in Makefile.am
as well as the AM_CONDITIONAL([SET_DLSCHED] .... in configure.ac

Reply:
Okay thanks. Done that. Just to confirm, all the `#ifdef DLSCHED` in the *.[ch] files should be removed now, right?

Reply:
And the configure.ac test is now shorter

Reply:
> Okay thanks. Done that. Just to confirm, all the `#ifdef DLSCHED` in the *.[ch] files should be removed now, right?
Yes.